commitizen/cmd.py

```diff
@@ -20,4 +22,10 @@ def run(cmd: str) -> Command:
     )
     stdout, stderr = process.communicate()
     return_code = process.returncode
-    return Command(stdout.decode(), stderr.decode(), stdout, stderr, return_code)
+    return Command(
+        stdout.decode(chardet.detect(stdout)["encoding"] or "utf-8"),
```

Comment:
What's the expected output of `chardet.detect` if it cannot detect anything? Will it be something like `{'encoding': None, 'confidence': 0.99}`?

Reply:
> What's the expected output of `chardet.detect` if it cannot detect anything? Will it be something like `{'encoding': None, 'confidence': 0.99}`?

Yes, it returns None, that's why I've added the `or` clause.

It also returns None for empty content `b''`, like the stderr content when all goes ok, so the `or` is mandatory to avoid `decode` failures.

Let me know if this "chardet-enabled" version is ok or if you prefer the previous one (with "iso..." encoding).

Reply:
just confirm it. I like the chatdet idea.